### PR TITLE
[1922] Shape Flow migration

### PR DESCRIPTION
**Pull request by mcharfadi**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/1922

### CODE REVIEW

File: doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+= (appetite) Shape
```
Comment (mbats) [inline doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc]: Give a name and an appetite to the shape

File: doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+== Problem
+
+In order to show that the view DSL can support everything the compatibility layer support, we should migrate flow to the view DSL. Using a dedicated package named flow, we should add a new project sirius-components-flow-view which should contain the view model for the various flow diagrams.
```
Comment (mbats) [inline doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc]: In order to include a test which covers the all odesign to view migration path, we will migrate the existing flow model.

File: doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+This view model should be generated using the soon to be introduced Java APIs to create view models. https://github.com/eclipse-sirius/sirius-components/issues/1919
+
+== Key Result
+
+The flow odesign model should be converted to a view model. All concepts used in the flow model should have a respective view concept.
+
+== Solution
+
+Use the the soon to be introduced Java APIs to create view models. https://github.com/eclipse-sirius/sirius-components/issues/1919
```
Comment (mbats) [inline doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc]: Use the the -> Use the

File: doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+=== Scenario
+
+Giving a flow model defined in a odesign file, we should be able to generate a view model witch match all concepts in the odesign model.
```
Comment (mbats) [inline doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc]: witch -> which
match all concepts -> match all the concepts existing in the odesign flow model

File: doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+=== Breadboarding
```
Comment (mbats) [inline doc/iterations/YYYY-MM/2023.6/shapes/shape Migration flow.adoc]: No UI involved

File: doc/iterations/2023.6/scopes/scope Migration flow.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+- [] First prototype of implementation
+- [] Improvement of the first prototype after the discovery of performance issues
+- [] Second prototype of the backend merged
+
+
+== Third task
+
+- [ ] User guide updated with screenshots
+- [ ] Integrated tests updated
```
Comment (sbegaudeau) [inline doc/iterations/2023.6/scopes/scope Migration flow.adoc:17]: As mentionned in your other shape pull request, do not contribute the scope now. It is to be maintained in your branch where you will work on the feature and contributed when it's over.
Same comment regarding the title of the commit / pull request

File: doc/iterations/2023.6/shapes/shape Migration flow.adoc [hunk 7]
```diff
@@ -0,0 +1,29 @@
+= (appetite) Shape
+
+== Problem
+
+In order to include a test which covers the all odesign to view migration path, we will migrate the existing flow model.
+
+This view model should be generated using the soon to be introduced Java APIs to create view models. https://github.com/eclipse-sirius/sirius-components/issues/1919
```
Comment (sbegaudeau) [inline doc/iterations/2023.6/shapes/shape Migration flow.adoc:7]: You need to specify clearly the order of the various tasks that you will perform during this iteration. It's not clear for me yet and there are multiple possible solutions.

File: doc/iterations/2023.6/shapes/shape Migration flow.adoc [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+== Key Result
+
+The flow odesign model should be converted to a view model. All concepts used in the flow model should have a respective view concept.
```
Comment (sbegaudeau) [inline doc/iterations/2023.6/shapes/shape Migration flow.adoc:11]: One sentence per file

File: doc/iterations/2023.6/shapes/shape Migration flow.adoc [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+== Solution
+
+Use the soon to be introduced Java APIs to create view models. https://github.com/eclipse-sirius/sirius-components/issues/1919
```
Comment (sbegaudeau) [inline doc/iterations/2023.6/shapes/shape Migration flow.adoc:15]: - Where will this file be located? in which project? In which Git repository? https://github.com/ObeoNetwork/FlowDesigner like the existing flow.odesign file? How can I reuse it in other applications using flow already?
- Will it depend on existing flow projects to retrieve the meta-model? (I think so) or will we migrate flow to domain too (that's definitely a no-go)
- Will it depend on existing flow projects to retrieve existing Java services? (I don't think so)

File: doc/iterations/2023.6/shapes/shape Migration flow.adoc [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+=== Scenario
+
+Giving a flow model defined in a odesign file, we should be able to generate a view model which  match all the concepts existing in the odesign flow model
```
Comment (sbegaudeau) [inline doc/iterations/2023.6/shapes/shape Migration flow.adoc:19]: The end user will not do that, the end user will open Sirius Web and not see and single change at all. Everything will work as before. including our Cypress integration tests which are heavily relying on flow.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Pull request closed without merging**